Update outdated docs, improve naming, prefer user props over auth hook in docs (#2535)
wasp-lang/wasp: web/docs/auth/overview.md

#### Using the `user` prop

> The preferred way to fetch user data on authenticated pages is by using the `user` prop. If the page's declaration sets `authRequired` to `true`, the page's React component receives the `user` object as a prop:

Reviewer: ```suggestion
If the page's declaration sets `authRequired` to `true`, the page's React component receives the `user` object as a prop. This is the simplest way to access the user inside an authenticated page:
```

</Tabs>

:::tip
While the `user` prop is only available in an authenticated page's React component, it is the preferred option because we can be certain that the user exists on that page. On the other hand, `useAuth` can be used when the user might not exist, making it better suited for non-authenticated components.

> :::

Reviewer: I'd move this to the beginning of the chapter. It's best to warn users about a better approach at the beginning.

And we can lose the tip - the text is part of the document's theme.

   The render function has the following signature:

>    ```

Reviewer 2: Maybe we can improve this code to be valid TS so we get syntax highlighting? That could be a nice improvement for readers.

Author: What would you write? 
If we want syntax highlighting we need something like:
```ts
type renderFn = (form: UseFormReturn, state: FormState) => React.ReactNode
```

But this may lead on readers that such type exists.

Reviewer 2: I'd write it like that:
```ts
type RenderFn = (form: UseFormReturn, state: FormState) => React.ReactNode
```
We are not writing an import statement, I think the readers would get it 👍

Author: Now that I've looked into the source code, we have this:
```ts
// PRIVATE API
export type AdditionalSignupFieldRenderFn = (
  hookForm: UseFormReturn<LoginSignupFormFields>,
  formState: FormState
) => React.ReactNode
```

Even though this is a private API I still think it would be preferable to include the same as the source code. It could be helpful to possible future contributors or anyone who wants to dig deeper into wasp. I would omit the `// PRIVATE API` comment and `export` for the docs.
@sodic you had some reservations around this, do you think this is fine?

Reviewer: Hm, maybe too much for my taste. I'd go with what you had originally (just the type signature) and maybe even inline it into the text.

That said, I am fine with whatever you like best.

One final piece of advice, ask yourself this: _"If I could have proper syntax highlighting (or none) for all options, which one would I pick?"_ And then go with that one.

We always want the best possible content, not the content that makes our tools the happiest :)

Anyway, pick whatever you like and feel free to resolve!
